src/app/mobile-app/middleware.ts

```diff
@@ -17,5 +17,13 @@ export const checkMobileAppDetails: RequestHandler = async (req, res) => {
     res.status(apiResponse.status);
   }
 
+  // If we don't have a clientOAuthSessionId in our session then we're dealing with a cross browser callback and core back will give us a clientOAuthSessionId to use instead.
+  if (
+    !req.session.clientOauthSessionId &&
```

Comment:
should this be if we don't have an ipvSessionId rather than clientOauthSessionId ?

Reply:
I'll add that as well. I'd like to keep this check so there's no way we can accidentally overwrite a clientOAuthSessionId